src/core/server/http/router/request.ts

```diff
@@ -33,10 +33,12 @@ const requestSymbol = Symbol('request');
  * Request specific route information exposed to a handler.
  * @public
  * */
-export interface KibanaRequestRoute {
+export interface KibanaRequestRoute<Method extends RouteMethod | 'patch' | 'options'> {
```

Comment:
Question @restrry: as they seems supported, why aren't `patch` and `options` actually in our `RouteMethod` type ?

Reply:
Probably because there are no use cases in the legacy platform. The initial implementation was done by @azasypkin

Reply:
Should we add them to `RouteMethod` then, or do you see any reason not to do so?

Reply:
I'd avoid extending API surface until it's necessary. YMMV